src/panels/flow-histogram-react/FlowHistogramHelpers.ts

```diff
@@ -215,3 +216,67 @@ export const getSeriesMetricValues = (fields: any[], name: string) => {
 
 }
 
+export const setLegend = (options: { flowHistogramOptions: FlowHistogramOptionsProps }) => {
+    if (options.flowHistogramOptions.showLegend) {
+        const className = options.flowHistogramOptions.position.label === 'Under Graph' ? '.flow-histogram-legend-bottom' : '.flow-histogram-legend-right'
+        const legend = $('.legend')
+        if (legend && legend.html()) {
+            $(className).html('')
+            $(className).append(legend.html())
+            legend.remove()
+        }
+    }
+}
+
+export const validateFlowHistogramPanelData = (dataSeries: DataFrame[]) => {
+    if (!dataSeries || !dataSeries[0].meta || !dataSeries[0].meta.custom) {
```

Comment:
Probably should be:
```
if (!dataSeries) {
  throw new Error('No data...')
} else if (!dataSeries[0].meta || !dataSeries[0].meta.custom) {
  throw new Error('Incorrect or incomplete data...')
}
...
```

Reply:
I believe if is not a flow datasource can have dataseries but not the meta or meta.custom